docs/proposals/metadata.md

```diff
@@ -0,0 +1,152 @@
+
+This variable contains a comma separated list of services that the Docker image
+publishes. The service is described as `<name>/PROTO/PORT` (eg.
+`mysql/tcp/3128`) where the 'PROTO' part is optional and defaults to 'tcp'.
+
+When the platform consumes this Docker image, it should automatically setup
+Service(s) with given name and port, the Route should be created for this
+service and the health-check should be generated for the ReplicationController
+that manages the Pods that make up this Service.
+
+Once this Docker image is imported into ImageRepository, the service might be
+referenced by other metadata (SUBSCRIBES, REQUIRES, see bellow.)
+
+
+### SUBSCRIBES
+
+This variable contains a list of services that the Docker image can consume.
+The value can be a valid Docker image name or the service name. In order to
+distinguish between them, the proposed format for the value us following:
+
+```
+SUBSCRIBES="service:redis/6379,image:openshift/redis"
+```
+
+The format of the 'service' value is the same as in PUBLISHES.
+
+In this case, we lookup user ImageRepositories for the Docker image that
+publishes `redis/6379` service. If there is no match, we continue to search for
+the `openshift/redis` image. If there is a match, the generation tools should
+automatically create corresponding Service and add an endpoint for this service
+based on the matching Docker image (iow. instantiate the image in a Pod).
+
+If there is no match, we inform the user that there is no image in the system
+that satisfy SUBSCRIBES, but we don't fail. The SUBSCRIBES is optional.
+
+//TODO: What if the service already exists in the namespace? Can a service be
+shared?
+
+### REQUIRES
+
+This variable contains a list of services that the Docker image requires in
+order to work properly. The value format is same as for SUBSCRIBES.  Where the
+SUBSCRIBES setup the infrastructure for pub/sub the REQUIRES might indicate that
+the Docker image requires another Docker image in order to work.  For example, a
+'phpmyadmin' requires 'mysql' service to be available, otherwise it won't work
+properly. A special 'backup' Docker image for mysql will require 'mysql' to be
+available within a Pod in order to do database dump from it.
+
+### SECRETS
+
+This variable contains a list of secrets (secret identifiers) that the Docker
+image requires to work properly. For example the Docker image might require
+Twitter API keys in order to send HTTP requests to the Twitter API. The
+generation tools should check if the required secrets are available and if not,
+it should ask the user to provide them.
+
+An example value can be: `SECRETS=keys/twitter,keys/facebook`
+
+### SINGLETON
+
+This variable indicates if the Docker image is clusterable or not. In case of
+Docker image with standalone 'mysql' server you don't want to have the number of
+replicas in ReplicationController be higher than 1 if the assumption is that the
+Service that mysql provides represents a single database.
```

Comment:
This is a suggestion, not a hard requirement.  The UIs will only suggest that you don't cluster this, they won't prevent you.